internal/exec/vendor_utils.go

```diff
@@ -120,48 +122,88 @@ func ExecuteVendorPullCommand(cmd *cobra.Command, args []string) error {
 }
 
 // ReadAndProcessVendorConfigFile reads and processes the Atmos vendoring config file `vendor.yaml`
-func ReadAndProcessVendorConfigFile(cliConfig schema.CliConfiguration, vendorConfigFile string) (
-	schema.AtmosVendorConfig,
-	bool,
-	string,
-	error,
-) {
+func ReadAndProcessVendorConfigFile(
+	cliConfig schema.CliConfiguration,
+	vendorConfigFile string,
+) (schema.AtmosVendorConfig, bool, string, error) {
 	var vendorConfig schema.AtmosVendorConfig
-	vendorConfigFileExists := true
+	var vendorConfigFileExists bool
+	var foundVendorConfigFile string
+
+	// Check if vendor config is specified in atmos.yaml
+	if cliConfig.Vendor.BasePath != "" {
+		if !filepath.IsAbs(cliConfig.Vendor.BasePath) {
+			foundVendorConfigFile = filepath.Join(cliConfig.BasePath, cliConfig.Vendor.BasePath)
+		} else {
+			foundVendorConfigFile = cliConfig.Vendor.BasePath
+		}
+	} else {
+		// Path is not defined in atmos.yaml, proceed with existing logic
+		var fileExists bool
+		foundVendorConfigFile, fileExists = u.SearchConfigFile(vendorConfigFile)
 
-	// Check if the vendoring manifest file exists
-	foundVendorConfigFile, fileExists := u.SearchConfigFile(vendorConfigFile)
+		if !fileExists {
+			// Look for the vendoring manifest in the directory pointed to by the `base_path` setting in `atmos.yaml`
+			pathToVendorConfig := path.Join(cliConfig.BasePath, vendorConfigFile)
 
-	if !fileExists {
-		// Look for the vendoring manifest in the directory pointed to by the `base_path` setting in the `atmos.yaml`
-		pathToVendorConfig := path.Join(cliConfig.BasePath, vendorConfigFile)
+			if !u.FileExists(pathToVendorConfig) {
+				vendorConfigFileExists = false
+				return vendorConfig, vendorConfigFileExists, "", fmt.Errorf("vendor config file or directory '%s' does not exist", pathToVendorConfig)
+			}
 
-		if !u.FileExists(pathToVendorConfig) {
-			vendorConfigFileExists = false
-			return vendorConfig, vendorConfigFileExists, "", fmt.Errorf("vendor config file '%s' does not exist", pathToVendorConfig)
+			foundVendorConfigFile = pathToVendorConfig
 		}
-
-		foundVendorConfigFile = pathToVendorConfig
 	}
 
-	vendorConfigFileContent, err := os.ReadFile(foundVendorConfigFile)
+	// Check if it's a directory
+	fileInfo, err := os.Stat(foundVendorConfigFile)
 	if err != nil {
-		return vendorConfig, vendorConfigFileExists, "", err
+		return vendorConfig, false, "", err
 	}
 
-	vendorConfig, err = u.UnmarshalYAML[schema.AtmosVendorConfig](string(vendorConfigFileContent))
-	if err != nil {
-		return vendorConfig, vendorConfigFileExists, "", err
+	var configFiles []string
+	if fileInfo.IsDir() {
+		// Read all .yaml files from directory
+		entries, err := os.ReadDir(foundVendorConfigFile)
```

Comment:
I think we should search for `*.yml` and `*.yaml` recursively.

Reply:
resolved using https://github.com/bmatcuk/doublestar